dissect/target/helpers/sunrpc/client.py

```diff
@@ -67,28 +76,94 @@ class IncompleteMessage(Exception):
     pass
 
 
-class Client(AbstractContextManager, Generic[Credentials, Verifier]):
+class InvalidPortMapping(Exception):
+    pass
+
+
+class AbstractClient(ABC):
+    @abstractmethod
+    def call(
+        self,
+        proc_desc: ProcedureDescriptor,
+        params: Params,
+        params_serializer: Serializer[Params],
+        result_deserializer: Deserializer[Results],
+    ) -> Results:
+        pass
+
+    @abstractmethod
+    def close(self) -> None:
+        pass
+
+    @abstractmethod
+    def rebind_auth(self, auth: AuthScheme[ConCredentials, ConVerifier]) -> None:
+        pass
+
+
+class FreePrivilegedPortType:
+    pass
+
+
+"""Marker type indicating Client should search for a free privileged port"""
+FreePrivilegedPort = FreePrivilegedPortType()
```

Comment:
Is this marker really necessary? I think since `0` is generally accepted as serving a similar purpose (albeit not for privileged ports), within the context of this API + with a docstring describing it, it's equally suitable for this purpose.

Or is there something I'm not seeing? I think in the code there's a difference between `0` and `FreePrivilegedPort`, but I don't really see it clearly documented. In any case, I think I'm more in favor of a more "Pythonic" approach to this, which is to use a "magic value" instead of a special marker object. For example `-1`, or `None`, and then clearly document those values in the docstrings.

Reply:
Maybe an `IntEnum` is better suited, which allows the "best of both worlds"? A named value, but also an integer value for those too lazy to import `FreePrivilegedPort` (e.g. when messing around in a quick script or REPL).